grakn-test/src/test/java/ai/grakn/test/GraphContext.java

```diff
@@ -0,0 +1,126 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.test;
+
+import ai.grakn.GraknGraph;
+import ai.grakn.engine.backgroundtasks.standalone.StandaloneTaskManager;
+import ai.grakn.engine.util.ConfigProperties;
+import ai.grakn.factory.GraphFactory;
+import org.junit.rules.ExternalResource;
+
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import static ai.grakn.engine.util.ConfigProperties.TASK_MANAGER_INSTANCE;
+import static ai.grakn.test.GraknTestEnv.ensureCassandraRunning;
+import static ai.grakn.test.GraknTestEnv.ensureHTTPRunning;
+import static ai.grakn.test.GraknTestEnv.usingOrientDB;
+import static ai.grakn.graphs.TestGraph.loadFromFile;
+import static ai.grakn.test.GraknTestEnv.usingTinker;
+
+/**
+ *
+ * @author alexandraorth
+ */
+public class GraphContext extends ExternalResource {
+
+    private GraknGraph graph;
+    private Consumer<GraknGraph> preLoad;
+    private String[] files;
+
+    private GraphContext(Consumer<GraknGraph> build, String[] files){
+        this.preLoad = build;
+        this.files = files;
+    }
+
+    public static GraphContext empty(){
+        return new GraphContext(null, null);
+    }
+
+    public static GraphContext preLoad(Consumer<GraknGraph> build){
+        return new GraphContext(build, null);
+    }
+
+    public static GraphContext preLoad(String... filesToLoad){
+        return new GraphContext(null, filesToLoad);
+    }
+
+    public GraknGraph graph(){
+        return graph;
+    }
+
+    public void rollback(){
+        try {
+            graph.rollback();
+        } catch (UnsupportedOperationException e) {
+            // If operation unsupported, make a fresh graph
+            closeGraph();
+            loadGraph();
+        }
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        ensureCassandraRunning();
+
+        //TODO remove when Bug #12029 fixed
+        ConfigProperties.getInstance().setConfigProperty(TASK_MANAGER_INSTANCE, StandaloneTaskManager.class.getName());
+        ensureHTTPRunning();
+        //TODO finish remove
+
+        // create the graph
+        loadGraph();
+    }
+
+    @Override
+    protected void after() {
+        closeGraph();
+    }
+
+    private void closeGraph(){
+        // close the graph
+        if(!graph.isClosed()) {
+            graph.clear();
+            graph.close();
+        }
+    }
+
+    private void loadGraph() {
+        String keyspace;
+        if (usingOrientDB()) {
+            keyspace = "memory";
+        } else {
+            // Embedded Casandra has problems dropping keyspaces that start with a number
+            keyspace = "a"+ UUID.randomUUID().toString().replaceAll("-", "");
+        }
+
+        graph = GraphFactory.getInstance().getGraph(keyspace);
```

Comment:
@aelred It uses a different factory, but I agree we can consolidate the "random keyspace name" bit.